src/lib.rs

```diff
@@ -317,6 +325,22 @@ impl Cascade {
         }
         false
     }
+
+    /// Determine the approximate amount of memory in bytes used by this
+    /// Cascade. Because this implementation does not integrate with the
+    /// allocator, it can't get an accurate measurement of how much memory it
+    /// uses. However, it can make a reasonable guess, assuming the sizes of
+    /// the bloom filters are large enough to dominate the overall allocated
+    /// size.
+    pub fn approximate_size(&self) -> usize {
+        self.filter.approximate_size()
+            + self
+                .child_layer
+                .as_ref()
+                .map_or(0, |child_layer| child_layer.approximate_size())
+            + self.salt.as_ref().map_or(0, |salt| salt.len())
+            + size_of::<bool>()
```

Comment:
and might as well throw in a `size_of::<Cascade>` for good measure.

Reply:
Right, so my understanding of rust is `size_of::<Cascade>()` will include `size_of::<Bloom>()` and `size_of::<BitVector>()`. Doing it this way simplifies things in a way, in that we can basically boil this all down to `size_of::<Cascade>()` plus the lengths of the filter bytes and salts.